internal/storage/v2/grpc/client.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/proto-gen/storage/v2"
+)
+
+type Client struct {
```

Comment:
Why not just TraceReader, externally `grpc.TraceReader`?

Reply:
also add
```
var _ tracestore.Reader = (*TraceReader)(nil)
```

Reply:
@yurishkuro Fixed. For the latter point, do you prefer to implement all the functions in this PR?

Reply:
no, they can throw errors for now